Sources/Core/Views/ProfileImageView/ProfileImageView.swift

```diff
@@ -0,0 +1,195 @@
+  @Published public var placeholder: Image?
+  @Published public var backgroundType = ProfileImageView.Background.plain(.clear)
+  @Published public var cornerRadius: ProfileImageView.CornerRadiusType = .circle
+  @Published public var contentMode: ContentMode = .fit
+  @Published public var borderColor: Color = .clear
+  @Published public var borderWidth: CGFloat = 0
+  @Published public var badging: ProfileImageView.BadgingMode = .none
+  @Published var url: URL?
+}
+
+@available(iOS 14, *)
+public struct ProfileImageView: View {
+  public var imageTapped: (() -> Void)?
+  public var badgeTapped: (() -> Void)?
+  @ObservedObject public var properties = ProfileImageProperties()
+  
+  public init() {}
+  
+  public var body: some View {
+    GeometryReader { geo in
+      ZStack(alignment: properties.badging.alignment) {
+        ImageView(properties: properties, profileTapped: triggerImageTapClosure)
+        BadgeView(style: properties.badging, size: .init(width: geo.size.width / 3.5, height: geo.size.height / 3.5), badgeTapped: tiggerBadgeTapClosure)
+      }
+    }
+  }
+  
+  private func triggerImageTapClosure() { imageTapped?() }
+  private func tiggerBadgeTapClosure() { badgeTapped?() }
+}
+
+@available(iOS 14, *)
+private extension ProfileImageView {
+  struct ImageView: View {
+    @ObservedObject var properties: ProfileImageProperties
+    var profileTapped: (() -> Void)
+    var optionalPlaceHolder: UIImage?
+    
+    var body: some View {
+      GeometryReader { geo in
+        constructProfileImage(placeholder: properties.placeholder)
+          .resolve(from: properties.url, placeholder: properties.placeholder)
+          .aspectRatio(contentMode: properties.contentMode)
+          .frame(width: geo.size.width, height: geo.size.height)
+          .gesture(TapGesture().onEnded({ profileTapped() }))
+          .background(getBackground())
+          .overlay(getShape(for: geo.size))
+          .cornerRadius(getCornerRadius(for: properties.cornerRadius))
+      }
+    }
+    
+    func constructProfileImage(placeholder: Image?) -> Image {
+      if let unwrappedPlaceholder = placeholder {
+        return unwrappedPlaceholder
+      }
+      
+      return Image(uiImage: UIImage())
+    }
+    
+    func getBackground() -> AnyView {
+      switch properties.backgroundType {
+      case .plain(let backgroundColor):
+        return AnyView(backgroundColor)
+      case let .linearGradient(gradient):
+        return AnyView(gradient)
+      case let .angularGradient(gradient):
+        return AnyView(gradient)
+      case let .radialGradient(gradient):
+        return AnyView(gradient)
+      }
+    }
+    
+    func getCornerRadius(for cornerType: ProfileImageView.CornerRadiusType) -> CGFloat {
+       switch cornerType {
+       case .circle:
+         return .infinity
+       case .custom(let cornerRadius):
+         return cornerRadius
+       }
+     }
+    
+    func getShape(for size: CGSize) -> AnyView {
```

Comment:
Maybe we should avoid using _AnyView_ if not needed (see here for reference: https://www.swiftbysundell.com/articles/avoiding-anyview-in-swiftui/)

For example, here we can use something like this:
```
@ViewBuilder
    var imageBackground: some View {
      switch properties.backgroundType {
      case .plain(let backgroundColor):
        backgroundColor
      case let .linearGradient(gradient):
        gradient
      case let .angularGradient(gradient):
        gradient
      case let .radialGradient(gradient):
        gradient
      }
    }
```